molecule/logger.py

```diff
@@ -19,11 +19,27 @@
 #  DEALINGS IN THE SOFTWARE.
 
 import logging
+import os
 import sys
 
 import colorama
 
-colorama.init(autoreset=True)
+
+def should_do_markup(file=sys.stdout):
+    if os.environ.get('PY_COLORS') == '1':
+        return True
+    if os.environ.get('PY_COLORS') == '0':
+        return False
+    return hasattr(file, 'isatty') and file.isatty() and \
+        os.environ.get('TERM') != 'dumb' \
+        and not (sys.platform.startswith('java') and os._name == 'nt')
+
+
+def colorama_init():
```

Comment:
Can we write a unit test directly against `colorama_init` and use https://docs.pytest.org/en/latest/reference.html#_pytest.monkeypatch.MonkeyPatch to mock the necessary environment values? You can also mock `file.isatty` to be sure also.